src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/ContainerCleaner.java

```diff
@@ -0,0 +1,59 @@
+package uk.gov.hmcts.reform.blobrouter.tasks.processors;
+
+import com.azure.storage.blob.BlobClient;
+import com.azure.storage.blob.BlobContainerClient;
+import com.azure.storage.blob.BlobServiceClient;
+import com.azure.storage.blob.models.BlobItem;
+import org.slf4j.Logger;
+import uk.gov.hmcts.reform.blobrouter.data.EnvelopeRepository;
+
+import java.util.Map;
+import java.util.UUID;
+
+import static java.util.stream.Collectors.toMap;
+import static org.slf4j.LoggerFactory.getLogger;
+import static uk.gov.hmcts.reform.blobrouter.data.model.Status.DISPATCHED;
+
+@SuppressWarnings("java:S1135") // ignore TODOs. will be removed when blob deletion part is implemented
+public class ContainerCleaner {
+
+    private static final Logger logger = getLogger(ContainerCleaner.class);
+
+    private final BlobServiceClient storageClient;
+    private final EnvelopeRepository envelopeRepository;
+
+    public ContainerCleaner(
+        BlobServiceClient storageClient,
+        EnvelopeRepository envelopeRepository
+    ) {
+        this.storageClient = storageClient;
+        this.envelopeRepository = envelopeRepository;
+    }
+
+    public void process(String containerName) {
+        logger.info("Deleting from container {}", containerName);
+
+        Map<String, UUID> envelopesToDelete = envelopeRepository.find(DISPATCHED, false)
+            .stream()
+            .collect(toMap(env -> env.fileName, env -> env.id));
+        BlobContainerClient containerClient = storageClient.getBlobContainerClient(containerName);
+
+        containerClient
+            .listBlobs()
+            .stream()
+            .map(BlobItem::getName)
+            .forEach(blobName -> {
+                tryToDeleteBlob(envelopesToDelete, containerClient, blobName);
+            });
+    }
+
+    private void tryToDeleteBlob(Map<String, UUID> envelopesToDelete, BlobContainerClient containerClient, String blobName) {
+        BlobClient blob = containerClient.getBlobClient(blobName);
+
+        if (envelopesToDelete.containsKey(blobName)) {
+            blob.delete();
```

Comment:
I guess we'll have to break the lease first. Otherwise, deletion of the blob might be impossible.

Reply:
On a second thought, it's fine. If the blob is leased, it's better to skip its processing.

Reply:
good point we would need to break the lease I think as the main job would be continuously running and chances are most of the time they would be leased.